dotnet/src/Connectors/Connectors.OpenAI/OpenAIPromptExecutionSettings.cs

```diff
@@ -22,66 +24,174 @@ public sealed class OpenAIPromptExecutionSettings : PromptExecutionSettings
     /// Default is 1.0.
     /// </summary>
     [JsonPropertyName("temperature")]
-    public double Temperature { get; set; } = 1;
+    public double Temperature
+    {
+        get => this._temperature;
+
+        set
+        {
+            if (this.IsFrozen)
+            {
+                throw new InvalidOperationException("OpenAIPromptExecutionSettings are frozen and Temperature cannot be modified.");
+            }
+            this._temperature = value;
+        }
+    }
 
     /// <summary>
     /// TopP controls the diversity of the completion.
     /// The higher the TopP, the more diverse the completion.
     /// Default is 1.0.
     /// </summary>
     [JsonPropertyName("top_p")]
-    public double TopP { get; set; } = 1;
+    public double TopP
+    {
+        get => this._topP;
+
+        set
+        {
+            if (this.IsFrozen)
+            {
+                throw new InvalidOperationException("OpenAIPromptExecutionSettings are frozen and TopP cannot be modified.");
```

Comment:
Nit: I don't think you need a different message per property. I suggest just defining a shared private ThrowIfFrozen helper.